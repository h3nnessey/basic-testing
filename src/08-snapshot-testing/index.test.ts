import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  test('should generate linked list from values 1', () => {
    expect(generateLinkedList(['a', 'b', 'c'])).toStrictEqual({
      value: 'a',
      next: {
        value: 'b',
        next: {
          value: 'c',
          next: {
            value: null,
            next: null,
          },
        },
      },
    });
  });

  test('should generate linked list from values 2', () => {
    // building a linked list even on the base of these few elements is time-consuming and extremely awkward
    const elements = ['d', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o'];

    expect(generateLinkedList(elements)).toMatchSnapshot();
  });
});
