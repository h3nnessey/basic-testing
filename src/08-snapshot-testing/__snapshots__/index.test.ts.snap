// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": {
          "next": {
            "next": {
              "next": {
                "next": {
                  "next": {
                    "next": {
                      "next": {
                        "next": null,
                        "value": null,
                      },
                      "value": "o",
                    },
                    "value": "n",
                  },
                  "value": "m",
                },
                "value": "l",
              },
              "value": "k",
            },
            "value": "j",
          },
          "value": "i",
        },
        "value": "h",
      },
      "value": "g",
    },
    "value": "f",
  },
  "value": "d",
}
`;
